OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -403,7 +407,18 @@ void RefreshServerListInner(List<GameServer> games)
 							continue;
 
 						var canJoin = game.IsJoinable;
+						var host = game.Address.Split(':')[0];
 						var item = ScrollItemWidget.Setup(serverTemplate, () => currentServer == game, () => SelectServer(game), () => Join(game));
+
+						var block = item.GetOrNull("LATENCY");
+						if (block != null)
+						{
+							var pingTask = pingCache.GetOrAdd(host, key => new Lazy<Task<double>>(() => Task.Run(() => OpenRA.Server.Server.GetAveragePingTime(key, 4)))).Value;
+							block.IsVisible = () => pingTask.IsCompleted;
+							block.Get<ColorBlockWidget>("LATENCY_COLOR").GetColor = () => LobbyUtils.LatencyColor(
+								new Session.ClientPing() { Latency = (long)pingTask.Result });
```

Comment:
This will create and discard roughly 600 ClientPings per second, which we obviously don't want to do.  This only needs to be evaluated once, when the task is complete.